resources/lta_cmd.py

```diff
@@ -0,0 +1,205 @@
+"""
+Command line utility for Long Term Archive.
+
+Run with `python -m resources.lta_cmd $@`.
+"""
+
+import argparse
+import asyncio
+import json
+import sys
+from typing import Dict
+
+from rest_tools.client import RestClient  # type: ignore
+
+from lta.config import from_environment
+
+EXPECTED_CONFIG = {
+    'LTA_REST_TOKEN': None,
+    'LTA_REST_URL': None
+}
+
+
+async def display_config(args) -> int:
+    if args.json:
+        print_dict_as_pretty_json(args.config)
+    else:
+        for key in args.config:
+            print(f"{key}:\t\t{args.config[key]}")
+    return 0
+
+
+def print_dict_as_pretty_json(d: Dict) -> None:
+    print(json.dumps(d, indent=4, sort_keys=True))
+
+
+async def request_ls(args) -> int:
+    rc = RestClient(args.config["LTA_REST_URL"], token=args.config["LTA_REST_TOKEN"])
+    try:
+        response = await rc.request("GET", "/TransferRequests")
+        if args.json:
+            print_dict_as_pretty_json(response)
+        else:
+            results = response["results"]
+            for x in results:
+                short_id = x["uuid"][:7]
+                create_time = x["create_timestamp"]
+                source_cook = x["source"].split(":")
+                path = source_cook[1]
+                source = source_cook[0]
+                dests = [y.split(":")[0] for y in x["dest"]]
+                print(f"{short_id}\t{create_time}\t{path}\t{source} -> {dests}")
+    except Exception as e:
+        print(e)
+        return -1
+    return 0
+
+
+async def request_new(args) -> int:
+    # configure a RestClient from the environment
+    rc = RestClient(args.config["LTA_REST_URL"], token=args.config["LTA_REST_TOKEN"])
+    # get some stuff
+    source = args.source
+    dest = args.dest
+    path = args.path
+    # construct the TransferRequest body
+    request_body = {
+        "source": f"{source}:{path}",
+        "dest": [f"{x}:{path}" for x in dest],
+    }
+    try:
+        response = await rc.request("POST", "/TransferRequests", request_body)
+        uuid = response["TransferRequest"]
+        tr = await rc.request("GET", f"/TransferRequests/{uuid}")
+        if args.json:
+            print_dict_as_pretty_json(tr)
+        else:
+            print(f"{uuid} - {tr['create_timestamp']}\n{source} -> {[x for x in args.dest]}\n{path}")
+    except Exception as e:
+        print(e)
+        return -1
+    return 0
+
+
+async def request_status(args) -> int:
+    rc = RestClient(args.config["LTA_REST_URL"], token=args.config["LTA_REST_TOKEN"])
+    try:
+        response = await rc.request("GET", "/TransferRequests")
+        results = response["results"]
+        for x in results:
+            if x["uuid"].startswith(args.uuid):
```

Comment:
This makes me wonder if we should make the uuid as an optional parameter to `/TransferRequests`.

Reply:
That would be the right way to implement this. I've opened an issue for it:
https://github.com/WIPACrepo/lta/issues/29

Unfortunately, even with it, I don't think it gets us away from the loop. The short ID (depending on how short it is, for example `"2"`) may return multiple objects, so we're still bound to iterating over the potential results.